src/plugins/manageQuickCss.tsx

```diff
@@ -0,0 +1,325 @@
+    }
+};
+
+const removeFromSnippetIdCache = (snippetId: string) => {
+    cachedSnippetIds = cachedSnippetIds.filter(id => id !== snippetId);
+};
+
+const importAllSnippets = async (snowflake: string, snippets: { snippetId: string; snippet: string; }[], strategy: AddStrategy) => {
+    for (const { snippetId, snippet } of snippets) {
+        await importCssSnippet(snippetId, snippet, strategy);
+    }
+};
+
+const removeAllSnippets = async (snippetIds: string[]) => {
+    for (const snippetId of snippetIds) {
+        await removeCssSnippet(snippetId);
+    }
+};
+
+const syncSnippetIds = async () => {
+    const quickCSS = await VencordNative.quickCss.get();
+    const snippetIds = await DataStore.get(STORE_KEY) || [];
+
+    const snippetIdsToRemove = snippetIds.filter((id: string) => !quickCSS.includes(id));
+
+    if (snippetIdsToRemove.length > 0) {
+        await DataStore.set(STORE_KEY, snippetIds.filter((id: string) => !snippetIdsToRemove.includes(id)));
+        await fetchSnippetIds();
+    }
+};
+
+const removeCssSnippet = async (snippetId: string) => {
+    let quickCss = await VencordNative.quickCss.get();
+
+    const regex = new RegExp(
+        String.raw`/\*\nsnippet ${snippetId}[^]*?/\* end snippet ${snippetId} \*/`,
+        "gs"
+    );
+
+    quickCss = quickCss.replace(regex, "");
+    await VencordNative.quickCss.set(quickCss);
+
+    const snippets = await DataStore.get(STORE_KEY);
+    await DataStore.set(STORE_KEY, snippets.filter((s: string) => s !== snippetId));
+
+    removeFromSnippetIdCache(snippetId);
+
+    Toasts.show({
+        message: "Removed QuickCSS snippet!",
+        type: Toasts.Type.SUCCESS,
+        id: Toasts.genId(),
+        options: {
+            duration: 2000,
+            position: Toasts.Position.BOTTOM,
+        },
+    });
+};
+
+const importCssSnippet = async (snippetId: string, snippet: string, strategy: AddStrategy) => {
+    const quickCss = await VencordNative.quickCss.get();
+
+    const existingCss = quickCss.trim();
+    const cleanedSnippet = snippet.trim();
+
+    let updatedCss: string;
+
+    switch (strategy) {
+        case AddStrategy.Replace:
+            updatedCss = cleanedSnippet;
+            break;
+        case AddStrategy.Append:
+            updatedCss = `${existingCss}\n\n${cleanedSnippet}`;
+            break;
+        case AddStrategy.Prepend:
+            updatedCss = `${cleanedSnippet}\n\n${existingCss}`;
+            break;
+    }
+
+    await VencordNative.quickCss.set(updatedCss);
+    await DataStore.set(STORE_KEY, [...cachedSnippetIds, snippetId]);
+
+    addToSnippetIdCache(snippetId);
+
+    Toasts.show({
+        message: "Imported QuickCSS snippet!",
+        type: Toasts.Type.SUCCESS,
+        id: Toasts.genId(),
+        options: {
+            duration: 2000,
+            position: Toasts.Position.BOTTOM,
+        },
+    });
+};
+
+const messageContextMenuPatch: NavContextMenuPatchCallback = (children, props) => () => {
```

Comment:
Unless you have a reason not to, why not destructure props right here? And type it to avoid bugs

```suggestion
const messageContextMenuPatch: NavContextMenuPatchCallback = (children, { message }: { message: Message }) => () => {
```